stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -2486,6 +3174,25 @@ impl NakamotoChainState {
             None
         };
 
+        // process the tenure-change if it happened, so that when block-processing begins, it happens in whatever the
+        // current tenure is
```

Comment:
I'm not sure it's significant, however, because once the miner produces the tenure-start block, and it gets signed, they'll immediately apply it to their chainstate before building the next block.  This will cause `advance_nakamoto_tenure` to get called.